### DISCUSSION TITLE
Can't make fcitx5 work with electron apps(chrome) on sway. gtk4 doesn't help.

### DISCUSSION THREAD

**Discussion by aca** [fcitx/fcitx5#536] [Q&A]
Hi, I'm trying with arch, sway. All works just fine except for electron apps.
I found that setting `gtk-version=4` is the solution. But for me, it just doesn't work. I had clean install os recently, so I don't think I messed up some configuration, but have no idea what to do. 

```
00:19 λ sway --version
sway version 1.7
00:19 λ fcitx5 --version
5.0.17
00:19 λ uname -a
Linux root 5.18.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 25 Jun 2022 20:22:01 +0000 x86_64 GNU/Linux
00:20 λ 
google-chrome-stable --gtk-version=4 --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform=wayland 
```

Thank you.

**Comment by wengxt** [comment]
Electron is kinda special from what I heard the changes in electron make it doesn't work with gtk4. I don't think it works with Gtk4 just like chrome. Use xwayland for electron, maybe, except its would be blurry if you enable scaling, otherwise it should be as good as X11.

If you don't care about position input window (the fcitx popup), e.g. you're using an IM engine doesn't require this, you could use chrome that way. Otherwise popup position for now would only possible to work for GNOME + https://github.com/wengxt/gnome-shell-extension-kimpanel case.

**Reply by wengxt** [reply]
With the chrome command above, I can at least typing with fcitx. I wonder if you set GTK_IM_MODULE=fcitx ? (install fcitx5-gtk)

If you are not sure, at least you could try:
```
GTK_IM_MODULE=fcitx google-chrome-stable --gtk-version=4 --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform=wayland
```
